### PR TITLE
Add docker builds

### PR DESCRIPTION
**Pull request by mslipper**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+workflows:
+  main:
+    jobs:
+      - build-dockerfile:
+          context:
+            - optimism
+          filters:
+            branches:
+              only:
+                - optimism-prototype
```
Comment (trianglesphere) [inline .circleci/config.yml:42]: Should we build on every branch though? It's needed to run the integration tests prior to merging into main.

Comment (mslipper) [reply .circleci/config.yml:42]: Sure, I can make it do that.

Comment (mslipper) [reply .circleci/config.yml:42]: Done. We should turn this off if we're making too many changes to this repo - will fill Docker Hub up fast.

**Review by trianglesphere — APPROVED** [review]
Will have to change tag here or in integration test (`latest` vs `develop`), but not a blocker.
LGTM

**Pull request merged**